code/include/playfab/AsyncProvider.h

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+#include <XAsyncProvider.h>
+#include <playfab/TaskQueue.h>
+#include <playfab/PlayFabError.h>
+#include <playfab/PlayFabUser.h>
+
+namespace PlayFab
+{
+
+// Provider is a base class that makes implementing XAsync API providers easier. It is designed to reduce the 
+// amount of boilerplate and edge cases each provider needs to implement.
+// 
+// Each XAsync API should define a derived provider class and override the relevant operations: Begin, DoWork,
+// and GetResult.
+class Provider

```

Comment:
was this a class brought over from XSAPI that has already been reviewed perhaps?

Reply:
It was developed for XSAPI/XAL, but funnily enough it was never actually integrated into those code bases.  It has been reviewed previously, however. 

---
In reply to: [595552747](https://github.com/jplafonta/XPlatCppSdk/pull/2#discussion_r595552747) [](ancestors = 595552747)